### ISSUE TITLE
gp_QuaternionSLerp doesnt return changed by-ref argument

### ISSUE THREAD

**Issue by jf---** [tpaviot/pythonocc-core#317]
```python
from OCC.gp import gp_Vec, gp_Quaternion, gp_QuaternionSLerp

vX = gp_Vec(12, 0, 0)
vY = gp_Vec(0, 12, 0)
v45 = (gp_Vec(1, 1, 1).Normalized() * 12)

q = gp_Quaternion()
q1 = gp_Quaternion(vX, vX)
q2 = gp_Quaternion(vX, vY)

interp = gp_QuaternionSLerp(q1, q2)

for i in range(100):
    i__ = i / 100.
    interp.Interpolate(i__, q)
    print("i: ", i__)
    print("quat X,Y,Z,W: {} / {} / {} / {}".format(q.X(), q.Y(), q.Z(), q.W()))


```

---

quat X,Y,Z,W: 0.0 / 0.0 / 0.0 / 1.0
i:  0.96
quat X,Y,Z,W: 0.0 / 0.0 / 0.0 / 1.0
i:  0.97
quat X,Y,Z,W: 0.0 / 0.0 / 0.0 / 1.0
i:  0.98
quat X,Y,Z,W: 0.0 / 0.0 / 0.0 / 1.0
i:  0.99
quat X,Y,Z,W: 0.0 / 0.0 / 0.0 / 1.0

looks like the by-ref argument `q` is not updated?

**Comment by tpaviot** [comment]
The python wrapper is incorrect:
```
void Interpolate (Standard_Real theT,gp_Quaternion  theResultQ);
```
should be
```
void Interpolate (Standard_Real theT,gp_Quaternion  &theResultQ);
```

**Comment by tpaviot** [comment]
Branch review/Quat-fix should fix the problem.

**Comment by tpaviot** [comment]
Fixed by PR #322
